tests/common.py

```diff
@@ -291,14 +293,29 @@ def mock_http_component_app(hass, api_password=None):
 
 
 @asyncio.coroutine
-def async_mock_mqtt_component(hass):
+def async_mock_mqtt_component(hass, config=None):
     """Mock the MQTT component."""
+    if config is None:
+        config = {mqtt.CONF_BROKER: 'mock-broker'}
+
+    @asyncio.coroutine
+    def async_subscribe(topic, msg_callback, qos, encoding):
+        @callback
+        def async_message_callback(msg_topic, payload, qos, retain):
+            msg = mqtt.Message(msg_topic, payload, qos, retain)
+            if not mqtt._match_topic(topic, msg.topic):
+                return
+            sub = mqtt.Subscription(topic, msg_callback, 0, encoding)
+            mqtt._run_subscription_callback(hass, sub, msg)
```

Comment:
(Using as much of the MQTT client code as possible here - this is still quite ugly though)